### ISSUE TITLE
[web] ListView shows lines between children when scrolling with mouse

### ISSUE THREAD

**Issue by iapicca** [flutter/flutter#64322]
## Steps to Reproduce

1. run the code below with the command `flutter run -d chrome` with the latest master

2. Use the mouse wheel or equivalent multi-touch gesture to scroll

**Expected results:** the ListView should not behave unexpectedly
and experience scrolling with gestures or mouse should be identical

**Actual results:**  the ListView unexpectedly should lines when scrolling with mouse wheel  or equivalent mult-itouch

<details>
  <summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp(key: Key('app')));

class MyApp extends StatelessWidget {
  const MyApp({Key key}) : super(key: key);
  @override
  Widget build(BuildContext context) => MaterialApp(
        home: const MyHomePage(key: Key('home')),
      );
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) => Material(
        child: ListView(
          shrinkWrap: true,
          children: <Widget>[
            for (int i = 0; i < 20; ++i)
              Container(
                height: 80,
                color: Colors.blue,
                child: const Center(
                  child: Text('hello'),
                ),
              ),
          ],
        ),
      );
}
```

</details>

<details>
  <summary>screenshot</summary>

![scroll issue](https://user-images.githubusercontent.com/45637986/90899993-688d9280-e3d1-11ea-8167-fef963cb5407.png)

</details>

<details>
  <summary>logs</summary>

```console
 francesco@francesco-yoga720  ~/projects/lines  flutter_master run -d chrome -v                                                                                                       ✔  2547  17:01:45 
[ +104 ms] executing: [/home/francesco/development/flutter_master/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +28 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] eef4220a878fd5dfe1551f76a78168347d6aa9eb
[   +1 ms] executing: [/home/francesco/development/flutter_master/] git tag --points-at HEAD
[  +11 ms] Exit code 0 from: git tag --points-at HEAD
[   +1 ms] executing: [/home/francesco/development/flutter_master/] git describe --match *.*.* --first-parent --long --tags
[  +25 ms] Exit code 0 from: git describe --match *.*.* --first-parent --long --tags
[        ] 1.21.0-9.0.pre-212-geef4220a8
[  +10 ms] executing: [/home/francesco/development/flutter_master/] git rev-parse --abbrev-ref --symbolic @{u}
[   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/home/francesco/development/flutter_master/] git ls-remote --get-url origin
[   +4 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +39 ms] executing: [/home/francesco/development/flutter_master/] git rev-parse --abbrev-ref HEAD
[   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[  +37 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +14 ms] executing: /usr/lib/Android/platform-tools/adb devices -l
[  +14 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +62 ms] executing: /usr/lib/Android/platform-tools/adb devices -l
[ +144 ms] Generating /home/francesco/projects/lines/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +74 ms] List of devices attached
[   +3 ms] List of devices attached
[  +52 ms] Launching lib/main.dart on Chrome in debug mode...
[  +46 ms] Updating assets
[  +89 ms] Syncing files to device Chrome...
[  +42 ms] Generating /home/francesco/projects/lines/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +42 ms] <- reset
[   +6 ms] /home/francesco/development/flutter_master/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/home/francesco/development/flutter_master/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/francesco/development/flutter_master/bin/cache/flutter_web_sdk/ --incremental
--target=dartdevc --debugger-module-names --experimental-emit-debug-metadata -Ddart.developer.causal_async_stacks=true --output-dill /tmp/flutter_tools.KFKZDH/flutter_tool.FBPCST/app.dill --libraries-spec
file:///home/francesco/development/flutter_master/bin/cache/flutter_web_sdk/libraries.json --packages .packages -Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --filesystem-root
/tmp/flutter_tools.KFKZDH/flutter_tools.GJUCTP --filesystem-scheme org-dartlang-app --initialize-from-dill build/cache.dill.track.dill --platform
file:///home/francesco/development/flutter_master/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill
[  +14 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+9965 ms] Syncing files to device Chrome... (completed in 10,062ms, longer than expected)
[   +1 ms] Synced 25.8MB.
[        ] <- accept
[        ] Caching compiled dill
[ +283 ms] [CHROME]:
[   +1 ms] [CHROME]:DevTools listening on ws://127.0.0.1:41961/devtools/browser/fd283427-0871-4fa2-b0c9-ab2bb324cafa
[+3579 ms] Debug service listening on ws://127.0.0.1:33425/6NJ9HxBzqd8=

[ +303 ms] Debug service listening on ws://127.0.0.1:33425/6NJ9HxBzqd8=
[   +1 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[        ] been thoroughly tested in production environments.
[        ] For more information see https://flutter.dev/web
[        ] 🔥  To hot restart changes while running, press "r" or "R".
[   +1 ms] For a more detailed help message, press "h". To quit, press "q".
[+13042 ms] Stopped debug service on ws://127.0.0.1:33425

[  +12 ms] Application finished.
[   +7 ms] "flutter run" took 27,833ms.
[   +3 ms] ensureAnalyticsSent: 0ms
[        ] Running shutdown hooks
[        ] Shutdown hook priority 4
[  +12 ms] Shutdown hooks complete
[   +2 ms] exiting with code 0
```

</details>

<details>
  <summary>doctor</summary>

```console
[✓] Flutter (Channel master, 1.21.0-10.0.pre.212, on Linux, locale en_US.UTF-8)
    • Flutter version 1.21.0-10.0.pre.212 at /home/francesco/development/flutter_master
    • Framework revision eef4220a87 (14 hours ago), 2020-08-20 17:46:07 -0700
    • Engine revision 4dc866283d
    • Dart version 2.10.0 (build 2.10.0-45.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /usr/lib/Android
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /usr/lib/Android
    • ANDROID_SDK_ROOT = /usr/lib/Android
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_265-8u265-b01-0ubuntu2~18.04-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✗] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1
    ✗ GTK 3.0 development libraries are required for Linux development.
      They are likely available from your distribution (e.g.: apt install libgtk-3-dev)
    ✗ The blkid development library is required for Linux development.
      It is likely available from your distribution (e.g.: apt install libblkid-dev)

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[!] VS Code
    • VS Code at /usr/share/code
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Linux (desktop)  • linux      • linux-x64      • Linux
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 84.0.4147.135

! Doctor found issues in 3 categories.
```

</details>

note: the issue occurs running `flutter run -d chrome --dart-define=FLUTTER_WEB_USE_SKIA=true`    as well

**Comment by iapicca** [comment]
possibly related to https://github.com/flutter/flutter/issues/40947

**Comment by ferhatb** [comment]
Note: The following reproduces subpixel rendering problem.

<div transform: translate(0px, 40.223px)>
   <div style="width: 782px; height: 80px; background-color: rgb(33, 150, 243);"></div></div>
<div transform: translate(0px, 120.223px)>
   <div style="width: 782px; height: 80px; background-color: rgb(33, 150, 243);"></div></div>

**Comment by caduandrade** [comment]
I have similar problems with layouts (#78475)

**Comment by HansMuller** [comment]
This looks like a known problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288

More info about the problem can be found here:
https://github.com/flutter/flutter/issues/14288#issuecomment-361375791
https://github.com/flutter/flutter/issues/17084#issuecomment-385718108
https://github.com/flutter/flutter/issues/15035#issuecomment-370028740

**Comment by huycozy** [comment]
This issue is reproducible on the latest `stable` and `master` channels. It happens on both `canvaskit` and `html` renderers.

<details>
<summary>Demo video</summary>


https://user-images.githubusercontent.com/104349824/217518756-e23f87cf-71f4-4ff8-8ecb-5f50461f0cfb.mov


</details>

<details>
<summary>Sample code (be compatible with the new flutter version)</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp(key: Key('app')));

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) => MaterialApp(
    home: const MyHomePage(key: Key('home')),
  );
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) => Material(
    child: ListView(
      shrinkWrap: true,
      children: <Widget>[
        for (int i = 0; i < 20; ++i)
          Container(
            height: 80,
            color: Colors.blue,
            child: const Center(
              child: Text('hello'),
            ),
          ),
      ],
    ),
  );
}
```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.7.1, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.1 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (11 hours ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Dolphin 2021.3.1.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio Dolphin 2021.3.1.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.2 19H218
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```

```bash
[!] Flutter (Channel master, 3.8.0-7.0.pre.23, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.8.0-7.0.pre.23 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6e7f580370 (6 minutes ago), 2023-02-08 12:37:24 +0800
    • Engine revision f310ffd146
    • Dart version 3.0.0 (build 3.0.0-204.0.dev)
    • DevTools version 2.21.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.75.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (4 available)
    • SM T225 (mobile)       • R9JT3004VRJ                          • android-arm64  • Android 13 (API 33)
    • iPhone 14 Pro (mobile) • AB2EBB55-6852-40A7-9A12-057128370920 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-1 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 110.0.5481.77

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

**Comment by iapicca** [comment]
> This looks like a known problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288
> 
> More info about the problem can be found here: [#14288 (comment)](https://github.com/flutter/flutter/issues/14288#issuecomment-361375791) [#17084 (comment)](https://github.com/flutter/flutter/issues/17084#issuecomment-385718108) [#15035 (comment)](https://github.com/flutter/flutter/issues/15035#issuecomment-370028740)


closing as duplicate of
- https://github.com/flutter/flutter/issues/14288
- https://github.com/flutter/flutter/issues/22056
